extensions/firestore-huggingface-inference-api/functions/src/tasks/question-answering.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  HfInference,
+  HfInferenceEndpoint,
+  QuestionAnsweringOutput,
+} from '@huggingface/inference';
+import { DocumentSnapshot } from 'firebase-admin/firestore';
+import config from '../config';
+
+export function questionAnswering(
+  snapshot: DocumentSnapshot,
+  inference: HfInference | HfInferenceEndpoint,
+): Promise<QuestionAnsweringOutput> {
+  const { question, context } = validateInput(snapshot.data());
+
+  const options = {
+    ...(inference instanceof HfInference && {
+      model: config.modelId,
+    }),
+    inputs: {
+      context,
+      question,
+    },
+  };
+
+  return inference.questionAnswering(options);
+}
+
+function validateInput(data: any) {
+  const { question, context } = data;
+
+  if (
+    !question ||
```

Comment:
Is the bool check for question & context needed here?

Reply:
yeah since both are required to make the request